### DISCUSSION TITLE
meep install for non-root: meep-python.cxx

### DISCUSSION THREAD

**Discussion by mesastrop** [NanoComp/meep#1918] [Q&A]
Hello -
Alas, I must build meep on an rh7 machine without root privileges. Using the minimum number of dependencies (python, libctl) I created a configure file, to build it in my local install directory,  but it seems the resulting makefile had meep-python.cxx: portion commented out so I got the no-rule error for it. Is there a flag for the configure so that meep-python will be made? Thanks.

**Comment by oskooi** [comment]
> it seems the resulting makefile had meep-python.cxx: portion commented out so I got the no-rule error for it.

[`meep-python.cxx`](https://github.com/NanoComp/meep/blob/master/python/Makefile.am#L110) is built as part of the Python API which is enabled *by default*. (To disable it, you need to specify `--without-python` as a compiler flag.) When building from source in a non-standard location, be sure to first read https://meep.readthedocs.io/en/latest/Build_From_Source/#paths-for-configuring which describes the necessary environment variables you need set to enable your compiler to find the libraries and header files.

**Comment by mesastrop** [comment]
Hello, actually, i didn’t want it disabled, and nevertheless after configuring with python 3 enabled, it was commented out in the make file. Is there another configure flag I’m missing?

Thanks, Martin

**Comment by mesastrop** [comment]
Hello sir,
The problem seems to occur in the Makefile in the python directory; lines 1402-1428 are commented out. These involve meep-python.cxx and mpb-python. Is MPB required? When I uncomment these lines it fails when looking for mpb.so.
I was a little confused as to all of the flags I needed for configure, but I followed the example shown in the documentation; this seemed to work for setting the location of the libctl library correctly after I build libctl. 

Thanks for your help

**Comment by stevengj** [comment]
You need SWIG installed to generate `meep-python.cxx`, and you need to configure with `--enable-maintainer-mode`, so probably you are missing one of these two.    

(Alternatively, you could use the [release `.tar.gz` file](https://github.com/NanoComp/meep/releases/tag/v1.22.0), which includes all of the pre-generated code like `configure` and `meep-python.cxx` files and does not require `--enable-maintainer-mode` … building directly from a `git clone` is more complicated.)

**Comment by mesastrop** [comment]
Thank you. What is maintainer-mode?
